challengeB07.java

```diff
@@ -0,0 +1,290 @@
+        return today.equals(specificDate);
+    }
+
+
+    // 6. Getting Current Time
+    public static void displayCurrentTime() {
+        LocalTime currentTime = LocalTime.now();
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm:ss");
+        String formattedTime = currentTime.format(timeFormatter);
+        System.out.println("Current Time: " + formattedTime);
+
+    }
+
+
+    // 7. Date Arithmetic
+    public static LocalDate addWeeksToToday(int weeks) {
+        LocalDate today = LocalDate.now();
+        LocalDate resultDate = today.plusWeeks(weeks);
+        return resultDate;
+    }
+
+
+    // 8. Weekday Identifier
+    public static String findDayOfWeek(String date) {
+        LocalDate parsedDate = LocalDate.parse(date);
+        String dayOfWeek = parsedDate.getDayOfWeek()
+                .getDisplayName(TextStyle.FULL, Locale.ENGLISH);
+        return dayOfWeek;
+    }
+
+
+    //        9. Interval Between Dates
+    public static long daysBetween() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Enter the first date (YYYY-MM-DD): ");
+        String firstDateInput = scanner.nextLine();
+
+        System.out.print("Enter the second date (YYYY-MM-DD): ");
+        String secondDateInput = scanner.nextLine();
+
+        LocalDate firstDate = LocalDate.parse(firstDateInput);
+        LocalDate secondDate = LocalDate.parse(secondDateInput);
+
+        long daysBetween = ChronoUnit.DAYS.between(firstDate, secondDate);
+        return daysBetween;
+
+    }
+
+
+
+    // 10. Date Authenticator
+    public class DateAuthenticator {
+        public static boolean isValidDate(String date) {
+            String[] parts = date.split("-");
+
+            if (parts.length != 3) {
+                return false;
+            }
+
+            try {
+                int year = Integer.parseInt(parts[0]);
+                int month = Integer.parseInt(parts[1]);
+                int day = Integer.parseInt(parts[2]);
+
+                if (month < 1 || month > 12) {
+                    return false;
+                }
+
+                int[] maxDaysInMonth = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+                if (month == 2 && isLeapYear(year)) {
+                    maxDaysInMonth[1] = 29; // February has 29 days in a leap year
+                }
+
+                if (day < 1 || day > maxDaysInMonth[month - 1]) {
+                    return false;
+                }
+
+                return true;
+            } catch (NumberFormatException e) {
+                return false;
+            }
+
+        }
+
+
+    }
+    // 10. Date Authenticator-metoda de verificare
+    public static boolean isLeapYear ( int year){
+        if (year % 4 == 0) {
+            if (year % 100 == 0) {
+                return year % 400 == 0;
+            } else {
+                return true;
+            }
+        } else {
+            return false;
+
+        }

```

Comment:
FYI este o metoda in LocalDate care face asta pt tine. 
`date.isLeapYear()` unde date este un LocalDate